src/windows/cppwinrt/LottieAnimationViewManager.cpp

```diff
@@ -0,0 +1,178 @@
+#include "pch.h"
+#include "LottieAnimationViewManager.h"
+
+#include <winrt/Windows.UI.Xaml.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
+#include <winrt/Windows.UI.Xaml.Media.h>
+#include <winrt/Microsoft.UI.Xaml.Controls.h>

```

Comment:
i like the direction this is going in! One thing to consider is that we are working on moving react native for windows on top of WinUI 3. For the time being we still have system xaml customer, obviously, so ideally we should get both scenarios working. The way we have done this in the core framework is by using some #defines and namespace aliases, e.g.
```
#ifdef USE_WINUI3
namespace xaml = winrt::Windows::UI::Xaml;
// ...
#else
namespace xaml = winrt::Microsoft::UI::Xaml;
// ...
#endif
```
(see other examples in the RNW code).
Would it be possible for you to do something similar in this project so that RNW apps that want to adopt WinUI3 are able to do so?

Reply:
Yeah, I can do that. What's the mechanism for surfacing the USE_WINUI3 flag to the consuming application? Should I be importing a specific property sheet into this library or is it taken care of by Microsoft.ReactNative.Uwp.CppLib.props / Microsoft.ReactNative.Uwp.CppLib.targets?

Reply:
thanks! the way we did it for RNW core was:
1. A build-time property called `UseWinUI3`. This is passed in by the build command line (or alternatively environment variable) when invoking msbuild (`msbuild /p:UseWinUI3=true ...`)
2. The vcxproj includes `$(ReactNativeWindowsDir)\PropertySheets\WinUI.props` (via `Microsoft.ReactNative.Uwp.CppLib.props`). This defines some variables and pushes a compiler preprocessor definition for `USE_WINUI3` so that code can then switch on `#ifdef USE_WINUI3`